pkg/nsx/services/inventory/node.go

```diff
@@ -57,3 +64,31 @@ func (s *InventoryService) initContainerClusterNode(clusterId string) error {
 	}
 	return nil
 }
+
+func (s *InventoryService) CleanStaleInventoryClusterNode() error {
+	log.Info("Clean stale InventoryClusterNode")
+	inventoryNodes := s.ClusterNodeStore.List()
+	for _, inventoryNode := range inventoryNodes {
+		inventoryNode := inventoryNode.(*containerinventory.ContainerClusterNode)
+
+		// Check if the associated Cluster exists
+		cluster := s.ClusterStore.GetByKey(inventoryNode.ContainerClusterId)
+		if cluster == nil {
```

Comment:
why need to check cluster? if no cluster, clusterNode can't be added.

Reply:
fixed